qutip/cy/brtools.pyx

```diff
@@ -403,7 +403,7 @@ cdef np.ndarray[complex, ndim=1, mode='c'] vec_to_fockbasis(double complex * eig
     PyDataMem_FREE(temp1)
     cdef np.ndarray[complex, ndim=1, mode='c'] out = \
                 np.PyArray_SimpleNewFromData(1, &nrows2, np.NPY_COMPLEX128, fock_vec)
-    PyArray_ENABLEFLAGS(out, np.NPY_OWNDATA)
+    PyArray_ENABLEFLAGS(out, np.NPY_ARRAY_OWNDATA)
```

Comment:
Could you elaborate a little on these changes? Did the numpy constant name change at some point or what happened?

Reply:
This is a deprecated C-API call as of Numpy 1.7 (years ago).  It added the `NPY_ARRAY_` namespace to tidy up their API.  The old ones are unlikely to be removed until Numpy 2.0 it seems, so it's not particularly urgent.

This is actually a commit I left in by accident in the rebase.  Originally I also wanted to specify the `NPY_DEPRECATED_APIVERSION=NPY_1_16_API_VERSION` macro in the compilation to keep us writing modern C-API code against Numpy.  Unfortunately, Cython 0.29 still generates some pre-1.7 API code when you use the `cdef ndarray[int, ndim=2]` form (instead of the more modern typed memoryview), so I had to remove it and I must have still picked this commit in the rebase.

I can force-push remove it, or I could just leave it in since it's in its own commit anyway, so anyone checking the git log will see it as a separate entity at least (caa5e32).

Reply:
Happy for it to stay in.

Reply:
Yeah, not ideal of me to mess up the PR history like that, but at least from the git user's perspective it doesn't matter so much.